mne/epochs.py

```diff
@@ -682,272 +799,47 @@ def plot_psd_topomap(self, bands=None, vmin=None, vmax=None, proj=False,
-                if on_missing == 'error':
-                    raise ValueError(msg)
-                elif on_missing == 'warning':
-                    logger.warning(msg)
-                    warnings.warn(msg)
-                else:  # on_missing == 'ignore':
-                    pass
-
-        # Select the desired events
-        values = list(self.event_id.values())
-        selected = in1d(events[:, 2], values)
-        self.events = events[selected]
-
-        n_events = len(self.events)
-        if n_events > 1:
-            if np.diff(self.events.astype(np.int64)[:, 0]).min() <= 0:
-                warnings.warn('The events passed to the Epochs constructor '
-                              'are not chronologically ordered.',
-                              RuntimeWarning)
-
-        if n_events > 0:
-            logger.info('%d matching events found' % n_events)
-        else:
-            raise ValueError('No desired events found.')
-
-        self.selection = np.where(selected)[0]
-        self.drop_log = []
-        for k in range(len(events)):
-            if events[k, 2] in values:
-                self.drop_log.append([])
-            else:
-                self.drop_log.append(['IGNORED'])
-
-        self.preload = preload
-        if self.preload:
-            self._data = self._get_data_from_disk()
-            self.raw = None
-        else:
-            self._data = None
-
-    def drop_bad_epochs(self):
+    def drop_bad_epochs(self, reject='original', flat='original'):
         """Drop bad epochs without retaining the epochs data.
 
         Should be used before slicing operations.
 
         .. Warning:: Operation is slow since all epochs have to be read from
-            disk. To avoid reading epochs form disk multiple times, initialize
+            disk. To avoid reading epochs from disk multiple times, initialize
             Epochs object with preload=True.
 
+        Parameters
+        ----------
+        reject : dict | str | None
+            Rejection parameters based on peak-to-peak amplitude.
+            Valid keys are 'grad' | 'mag' | 'eeg' | 'eog' | 'ecg'.
+            If reject is None then no rejection is done. If 'original',
+            then the rejection parameters set at instantiation are used.
+        flat : dict | str | None
+            Rejection parameters based on flatness of signal.
+            Valid keys are 'grad' | 'mag' | 'eeg' | 'eog' | 'ecg', and values
+            are floats that set the minimum acceptable peak-to-peak amplitude.
+            If flat is None then no rejection is done. If 'original',
+            then the flat parameters set at instantiation are used.
+
+        Notes
+        -----
+        Dropping bad epochs can be done multiple times with different
+        ``reject`` and ``flat`` parameters. However, once an epoch is
+        dropped, it is dropped forever, so if more lenient thresholds may
+        subsequently be applied, `epochs.copy` should be used.
```

Comment:
But is it supposed to also have the functionality I mention above, namely allowing instantiating with `reject=flat=None`, and then at some later point applying some criteria? I guess I'm still not clear if the goal is just to 1) sanity check values like you say, or also 2) allow instantiating with `None` and doing rejection later.

I had assumed that if a user wants to know if they have rejected, they can compare `epochs.reject` to whatever they expect their rejection threshold to be. Is the idea to have `drop_bad_epochs` do that for them, then? If we don't want the instantiate-with-None-reject-later-with-something functionality mentioned above, then it wouldn't make sense to add arguments to `drop_bad_epochs`. 

FWIW the version I have programmed here essentially does the same thing that @dengemann's gist does from #1893.

I'm also confused because the re-calling thing seems to be new, and not what was discussed in #1900 (@mainakjas followed by @agramfort comments):

>>  so we throw an error also in the case that the user calls drop_bad_epochs
>> with the same reject and flat parameters?
>
> yes. It should not happen in real life for me

So it sounds like from before that dropping should only be allowed exactly once, and that subsequent calls to `drop_bad_epochs` with any arguments should be an error. What you are arguing for is different now, right, in that subsequent calls should throw an error only if the arguments don't match?